docs/modules/ROOT/pages/network/cilium-cluster-mesh.adoc

```diff
@@ -0,0 +1,139 @@
+<3> The field `spec.ports` is relevant on clusters which aren't hosting the workload.
+The values of `targetPort` and `protocol` for each entry of `spec.ports` must match the workload's container ports.
+The value of `port` can be chosen arbitrarily and doesn't need to match across clusters.
+<4> The example assumes that the service pods (and therefore the endpoints) have label `app=global-service`.
+The selector only matters on the cluster which hosts the workload.
+
+With that setup, each cluster can then discover the remote service via DNS name `global-service.namespace.svc.cluster.local`.
+
+NOTE: VSHN Managed OpenShift currently doesn't support cross-cluster service discovery via in-cluster DNS.
+
+To ensure that clients on all clusters can access the endpoints of the global service a network policy is required.
+Depending on the required level of isolation, a policy could allow all clusters to access the endpoints of the service by setting up a network policy which allows all traffic from all clusters:
+
+[source,yaml]
+----
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: allow-service-from-all-namespaces-all-clusters
+  namespace: backend <1>
+spec:
+  ingress:
+  - from:
+    - namespaceSelector: {} <2>
+  podSelector:
+    matchLabels:
+      app: global-service <3>
+  policyTypes:
+  - Ingress
+----
+<1> This example assumes that the global service is deployed in namespace `backend`.
+The NetworkPolicy must be deployed in the service namespace on the cluster hosting the service.
+<2> Allow traffic from all namespaces across all clusters.
+Omitting this will only allow traffic from namespace `backend`.
+<4> The example assumes that the service pods (and therefore the endpoints) have label `app=global-service`.
+
+See section xref:network/try-cluster-mesh.adoc#_remote_service_access_from_a_different_namespace["Remote service access from a different namespace"] for a hands-on cross-namespace remote access example.
+
+[NOTE]
+====
+If you know the pod IP of an application on a remote cluster, you can access that IP directly via the Cilium SDN when Cluster Mesh is setup.
+However, service IPs on a remote cluster aren't reachable through the Cilium SDN even with Cilium Cluster Mesh.
+See section xref:network/try-cluster-mesh.adoc#_accessing_a_remote_pod_directly_via_pod_ip["Accessing a remote pod directly via Pod IP"] for a hands-on example.
+====
+
+== Network Policies
+
+Cilium's network policy implementation treats each namespace name as a shared virtual namespace across all clusters which are connected in a Cluster Mesh.
+To restrict access between namespaces with the same name on different clusters, network policies must restrict access based on the label `io.cilium.k8s.policy.cluster`.
+VSHN Managed OpenShift uses each cluster's Project Syn ID as the Cilium Cluster Mesh cluster name.
```

Comment:
Can we add a how-to for how to discover this ID?

Reply:
I couldn't think of a simple way to discover the ID within the cluster without having cluster-wide permissions and without setting up a new object which is accessible for all authenticated users (with cluster-wide permissions I'm sure we could find an object that always exists and contains the cluster ID in a well-known field).

Other places where the ID may exists are the API/console URL, but not for all clusters. On clusters use cluster-mesh cross cluster isolation the ID is discoverable by looking at the default network policies.

Additionally, some customers already know/use the Project Syn IDs of their clusters when referring to the clusters.